src/main.c

```diff
@@ -505,18 +519,73 @@ static const struct xdg_toplevel_listener xdg_toplevel_listener = {
 	.close = xdg_toplevel_close,
 };
 
+static void wl_surface_enter(void* data, struct wl_surface* wl_surface,
+		struct wl_output* wl_output)
+{
+	struct window * w = data;
+	struct output* out;
+
+	wl_list_for_each(out, &outputs, link) {
```

Comment:
Maybe it would be nicer to use a function called something like `output_find_by_wl_output` here.